src/main/java/com/google/sps/servlets/ExamServlet.java

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.flogger.FluentLogger;
+import com.google.sps.data.UtilityClass;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet to display the requested exam  and send responses to ExamResponseServlet or if no exam
+ * is selected it will display a list of available exams.
+ *
+ * @author Aidan Molloy
+ */
+@WebServlet("/exam")
+public class ExamServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+  /**
+   * Gets the exam ID from the httpRequest, displays exam if it exists otherwise displays
+   * list of exams.
+   *
+   * @param request  provides request information from the HTTP servlet
+   * @param response response object where servlet will write information to
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Only logged in users should access this page.
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      logger.atWarning().log("User is not logged in.");
+      response.sendRedirect("/");
+      return;
+    }
+    logger.atInfo().log("user=%s", userService.getCurrentUser());
+    response.setContentType("text/html;");
+    PrintWriter out = response.getWriter();
+    out.println("<!DOCTYPE html>");
```

Comment:
if you switch to Java 12, you could use a multiline string literal.

Also based on a quick search it seems that you can embed a plain text file into .jar and then use "getClass().getClassLoader().getResourceAsStream("filename.txt");" to access its contents. This would allow you to have a .html file in the project, which would make it a bit more readable.

Reply:
Issue created referencing this.